custom_components/flashforge_adventurer_3/sensor.py

```diff
@@ -14,74 +14,91 @@
 )
 import voluptuous as vol
 
-from .const import DOMAIN
+from .const import CONF_PRINTERS, DEFAULT_PORT, DOMAIN, CONF_MODEL, AVAILABLE_MODELS
 from .protocol import get_print_job_status
 
 LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Required('ip'): cv.string,
+        vol.Required('ip_address'): cv.string,
         vol.Required('port'): cv.string,
+        vol.Required(CONF_MODEL): cv.string,
     }
 )
 
 class PrinterDefinition(TypedDict):
+    model: str
     ip: str
     port: int
 
-
 async def async_setup_entry(
-    hass: core.HomeAssistant,
+    hass: core.HomeAssistant, 
     config_entry: config_entries.ConfigEntry,
-    async_add_entities: Callable,
+    async_add_entities: Callable
 ) -> bool:
+
+    logger = logging.getLogger(__name__)
+    logger.info("Setting up sensor platform for entry %s", config_entry.entry_id)
+    
     config = hass.data[DOMAIN][config_entry.entry_id]
-    if config_entry.options:
-        config.update(config_entry.options)
-    coordinator = FlashforgeAdventurer3Coordinator(hass, config)
+
+    logger.debug("Config: %s", config)
+    
+    coordinator = FlashForgeAdventurer3Coordinator(hass, config)
+    logger.info("Coordinator created")
+    
     await coordinator.async_config_entry_first_refresh()
+    logger.info("First refresh of coordinator done")
+
     sensors = [
-        FlashforgeAdventurer3StateSensor(coordinator, config),
-        FlashforgeAdventurer3ProgressSensor(coordinator, config),
+        FlashForgeAdventurer3StateSensor(coordinator, config),
+        FlashForgeAdventurer3ProgressSensor(coordinator, config),
+        FlashForgeAdventurer5TemperatureSensor (coordinator, config),
     ]
+
+    logger.debug("Sensors list: %s", sensors)
+    
     async_add_entities(sensors, update_before_add=True)
+    logger.info("Added sensor entities")
 
 
-class FlashforgeAdventurer3Coordinator(DataUpdateCoordinator):
+class FlashForgeAdventurer3Coordinator(DataUpdateCoordinator):
     def __init__(self, hass, printer_definition: PrinterDefinition):
         super().__init__(
             hass,
             LOGGER,
             name='My sensor',
             update_interval=timedelta(seconds=60),
         )
-        self.ip = printer_definition['ip_address']
-        self.port = printer_definition['port']
+        self.ip = printer_definition.get('ip_address', 'Unknown IP')  # Set a default IP if not provided
+        self.port = printer_definition.get('port', 8899)  # Set a default port if not provided
+        self.model = printer_definition.get(CONF_MODEL, 'FlashForge Adventurer 3')
+
```

Comment:
🤔 I think it'll be better for the integration to fail early, rather than try to telnet to `Unknown IP`.